packaging/fedora/snapd.spec

```diff
@@ -72,6 +72,11 @@
 %define gobuild_static(o:) go build -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-}" -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -linkmode external -extldflags '%__global_ldflags -static'" -a -v -x %{?**};
 %endif
 
+# These macros are missing BUILDTAGS in RHEL 8
+%if 0%{?rhel} == 8
+%define gobuild(o:) go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-}" -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags'" -a -v -x %{?**};
```

Comment:
Pretty sure thus should already be part of go rpm macros. Let me double check.

Reply:
I see failures in centos8 currently that indicate that the testkeys are not compiled into the binary making some of our spread tests fail. This is why I suspect it's missing. But please do double check.

Reply:
Yeah, the macros in Fedora are ready for build tags, ones in CentOS are not.